src/ServiceControl/Infrastructure/Extensions/WaitHandleExtensions.cs

```diff
@@ -22,7 +22,7 @@ internal static class WaitHandleExtensions
                 tokenRegistration = cancellationToken.Register(
                     state => ((TaskCompletionSource<bool>)state).TrySetCanceled(),
                     tcs);
-                return await tcs.Task;
```

Comment:
This change has to be reverted because it changes the semantics of the code. Now the finally block is executed directly and registrations are disposed. We only want to execute the finally block after the task is done

Reply:
that would only be the case if you don't await `WaitOneAsync` though?

Reply:
Nope. Similar code like

```
using(var disposable) {
  return await Foo();
}
```

vs

```
using(var disposable) {
  return Foo();
}
```

First example disposable is disposed after Foo is done. Second example disposable is disposed immediately after Foo reached internal continuation but might not be done yet.

Reply:
ah, I didn't notice the finally block :( Reverting that change then.

Reply:
This is in combination with the other change below might cause that many test timeouts on the ATTs (see build server output)